### PR TITLE
[GEN-1724]: create a component for source containers info (display in drawer)

### PR DESCRIPTION
**Pull request by BenElferink**
This pull request includes significant updates to the frontend codebase, focusing on improving the handling of programming language icons and adding a new component for container details. The key changes involve refactoring utility functions, updating component imports, and introducing new styled components.

### Improvements to programming language handling:
* [`frontend/webapp/containers/main/sources/source-drawer-container/index.tsx`](diffhunk://#diff-d4745da2f63be90cb1f7af8ea29fb2f37cf0a80f0f04a207b6fc9149855ca6d5L9-R12): Replaced `getMainContainerLanguageLogo` with `getProgrammingLanguageIcon` for better icon handling and added `ContainerDetails` component to display detected containers. [[1]](diffhunk://#diff-d4745da2f63be90cb1f7af8ea29fb2f37cf0a80f0f04a207b6fc9149855ca6d5L9-R12) [[2]](diffhunk://#diff-d4745da2f63be90cb1f7af8ea29fb2f37cf0a80f0f04a207b6fc9149855ca6d5L105-R105) [[3]](diffhunk://#diff-d4745da2f63be90cb1f7af8ea29fb2f37cf0a80f0f04a207b6fc9149855ca6d5R127)
* [`frontend/webapp/utils/functions/icons.ts`](diffhunk://#diff-869883e9c9018f4ea6797d1fd860989852023765e868dcdee9fc50d72ad667c2R48-R68): Added `getProgrammingLanguageIcon` function to fetch programming language icons based on the language.

### New component for container details:
* [`frontend/webapp/reuseable-components/container-details/index.tsx`](diffhunk://#diff-efc30c3f1c39f706aa0796927eac89e9cee190ea773e9e30d8d611cd30e81b4aR1-R123): Introduced `ContainerDetails` component to display details of detected containers, including their names, languages, and instrumentation status.
* [`frontend/webapp/reuseable-components/index.ts`](diffhunk://#diff-ab2c0f4825fb371714af6d7da808d6a9ed85b1b0d55135316364fb5500c1f1f8R36): Exported `ContainerDetails` from reusable components.

### Refactoring and cleanup:
* [`frontend/webapp/utils/constants/programming-languages.ts`](diffhunk://#diff-214a28f6c56d77642ca8d1a14cbb3bf89b14a33da8fdb4506c76a6533bfea0a4L3-L4): Removed `LANGUAGES_LOGOS` and `LANGUAGES_COLORS` constants and the `getMainContainerLanguageLogo` function, consolidating icon handling into `getProgrammingLanguageIcon`. [[1]](diffhunk://#diff-214a28f6c56d77642ca8d1a14cbb3bf89b14a33da8fdb4506c76a6533bfea0a4L3-L4) [[2]](diffhunk://#diff-214a28f6c56d77642ca8d1a14cbb3bf89b14a33da8fdb4506c76a6533bfea0a4L22-R20) [[3]](diffhunk://#diff-214a28f6c56d77642ca8d1a14cbb3bf89b14a33da8fdb4506c76a6533bfea0a4L70-L92)
* [`frontend/webapp/types/sources.ts`](diffhunk://#diff-2e094683c1e156a4d3b841453d12f98342db28ae37230264200cf979063268b9L1-R6): Updated `SourceContainer` type to use `WORKLOAD_PROGRAMMING_LANGUAGES` enum.

These changes collectively enhance the frontend's ability to handle and display programming language information more efficiently and add new functionality for container details visualization.

### CODE REVIEW

File: frontend/webapp/reuseable-components/container-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+              <RowBody>
+                <Text size={14}> {containerName}</Text>
+                <Text size={10} color={theme.text.grey} style={{ textTransform: 'capitalize' }}>
+                  {language}
+                  <span style={{ textTransform: 'none' }}> • Runtime: {runtimeVersion}</span>
+                </Text>
+              </RowBody>
+
+              <InstrumentStatus $active={active}>
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/container-details/index.tsx:111]: with error icon is not part of the status func?

- container is uninstrumented if the programming language is "igonred" or "unknow" (double check this in the old UI code)

File: frontend/webapp/reuseable-components/container-details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  padding: 2px 6px;
+  margin-left: auto;
+  border-radius: 360px;
+  border: 1px solid ${({ $active, theme }) => ($active ? theme.colors.dark_green : theme.colors.border)};
+`;
+
+export const ContainerDetails: React.FC<Props> = ({ containers }) => {
+  return (
+    <Container>
+      <Header>
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/container-details/index.tsx:85]: dont we have reusable component for the card? as well as for the list item?

File: frontend/webapp/reuseable-components/container-details/index.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                  {language}
+                  <span style={{ textTransform: 'none' }}> • Runtime: {runtimeVersion}</span>
+                </Text>
+              </RowBody>
+
+              <InstrumentStatus $active={active}>
+                <Image src={active ? getStatusIcon('success') : '/icons/common/circled-cross.svg'} alt='' width={12} height={12} />
+                <Text size={10} family='secondary' color={active ? theme.text.success : theme.text.grey}>
+                  {active ? 'Instrumented' : 'Uninstrumented'}
+                </Text>
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/container-details/index.tsx:115]: let's start user string as const from single file

File: frontend/webapp/styles/theme.ts [hunk 4]
```diff
@@ -104,6 +105,20 @@ export const hexPercentValues = {
   '000': '00',
 };
 
+export const languageColors: Record<WORKLOAD_PROGRAMMING_LANGUAGES, string> = {
```
Comment (alonkeyval) [inline frontend/webapp/styles/theme.ts:108]: im not sure we need this anymore

File: frontend/webapp/types/sources.ts [hunk 5]
```diff
@@ -1,8 +1,11 @@
-import { Condition } from './common';
+import { type Condition } from './common';
+import { WORKLOAD_PROGRAMMING_LANGUAGES } from '@/utils';
 
 export type SourceContainer = {
   containerName: string;
-  language: string;
+  language: WORKLOAD_PROGRAMMING_LANGUAGES | 'ignored';
```
Comment (alonkeyval) [inline frontend/webapp/types/sources.ts:6]: why ignored not inside the object?

File: frontend/webapp/reuseable-components/container-details/index.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            <Row key={`container-${idx}`}>
+              <LanguageIcon>
+                <Image src={getProgrammingLanguageIcon(language)} width={20} height={20} alt='source' />
+              </LanguageIcon>
+
+              <RowBody>
+                <Text size={14}> {containerName}</Text>
+                <Text size={10} color={theme.text.grey} style={{ textTransform: 'capitalize' }}>
+                  {language}
+                  <span style={{ textTransform: 'none' }}> • Runtime: {runtimeVersion}</span>
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/container-details/index.tsx:107]: textTransform is not "none" on default?

File: frontend/webapp/reuseable-components/container-details/index.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+          return (
+            <Row key={`container-${idx}`}>
+              <LanguageIcon>
+                <Image src={getProgrammingLanguageIcon(language)} width={20} height={20} alt='source' />
+              </LanguageIcon>
+
+              <RowBody>
+                <Text size={14}> {containerName}</Text>
+                <Text size={10} color={theme.text.grey} style={{ textTransform: 'capitalize' }}>
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/container-details/index.tsx:105]: this is capitalize in Figma as well?

Comment (BenElferink) [reply frontend/webapp/reuseable-components/container-details/index.tsx:105]: Yes it is :)
`capitalize` makes only the 1st letter an uppercase, unlike `uppercase` which does the whole thing.

File: frontend/webapp/reuseable-components/container-details/index.tsx (hunk 2 above)
Comment (BenElferink) [reply frontend/webapp/reuseable-components/container-details/index.tsx:85]: No, `CardDetails` is a very different component than `ContainerDetails`, the only thing that is similar is the wrapper, none of the contents are the same.

File: frontend/webapp/reuseable-components/container-details/index.tsx (hunk 1 above)
Comment (BenElferink) [reply frontend/webapp/reuseable-components/container-details/index.tsx:111]: Applied logic for "uninstrumented" 👍

As for the icon, it's not an error, but a unique icon called "circled cross", I'll be making a function later for general icons that are not part of a category.

**Review by alonkeyval — APPROVED** [review]

**Pull request merged**